### PR TITLE
consensus: replace reth-primitive imports

### PR DESCRIPTION
**Pull request by estensen**
Progress on https://github.com/paradigmxyz/reth/issues/10749

Changed all where direct drop-in-replacement was possible

### CODE REVIEW

File: crates/consensus/beacon/Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ repository.workspace = true
 workspace = true
 
 [dependencies]
+alloy-primitives.workspace = true
```
Comment (mattsse) [inline crates/consensus/beacon/Cargo.toml:14]: pedantic nit, this should be moved below reth deps

File: crates/consensus/common/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,7 @@ repository.workspace = true
 workspace = true
 
 [dependencies]
+alloy-primitives.workspace = true
```
Comment (mattsse) [inline crates/consensus/common/Cargo.toml:14]: same

**Review by mattsse — DISMISSED** [review]
ty,

nits

File: crates/consensus/beacon/Cargo.toml (hunk 1 above)
Comment (estensen) [reply crates/consensus/beacon/Cargo.toml]: done

**Review by onbjerg — APPROVED** [review]

**Pull request merged**